cloudtasks/pull_queue/main.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+
+	"golang.org/x/net/context"
+
+	cloudtasks "cloud.google.com/go/cloudtasks/apiv2beta2"
+	duration "github.com/golang/protobuf/ptypes/duration"
+	taskspb "google.golang.org/genproto/googleapis/cloud/tasks/v2beta2"
+)
+
+// usage powers the help documentation when a help operation is used or the wrong number of arguments specified.
+func usage() {
+	fmt.Println("Usage of tasks_cli:")
+	fmt.Println()
+	fmt.Println("\t$> tasks create $PROJECT_ID $LOCATION_ID $QUEUE_ID")
+	fmt.Println("\t$> tasks pull $PROJECT_ID $LOCATION_ID $QUEUE_ID")
+	fmt.Println("\t$> tasks help")
+	fmt.Println("\nFor more information, see https://cloud.google.com/cloud-tasks/docs")
+}
+
+func main() {
+	args := os.Args[1:]
+	if len(args) == 0 || (len(args) < 3 && args[0] != "help") {
+		usage()
+		os.Exit(1)
+	}
+
+	switch args[0] {
+	case "create":
+		runTaskCreate(args[1:])
+	case "pull":
+		runTaskLeaseAndAck(args[1:])
+	default:
+		usage()
+	}
+
+	fmt.Println()
+}
+
+// runTaskCreate is invoked by the CLI for the "create" operation.
+func runTaskCreate(args []string) {
+	_, err := taskCreate(args[0], args[1], args[2])
+	if err != nil {
+		log.Fatalf("Error creating task: %s\n", err)
+	}
+}
+
+// runTaskPull is invoked by the CLI for the "pull" operation.
+func runTaskLeaseAndAck(args []string) {
+	task, err := taskLease(args[0], args[1], args[2])
+	if err != nil {
+		log.Fatalf("Error leasing task: %s\n", err)
+	} else if task == nil {
+		log.Println("No tasks available for lease")
+	} else if taskAck(task) != nil {
+		log.Fatalf("Error acknowledging task: %s\n", err)
+	}
+}
+
+// [START cloud_tasks_create_task]
```

Comment:
Please add a newline between region tags and the function documentation. Otherwise, the tag ends up in the GoDoc.